File: src/PAR.Domain/Entities/SchoolYear.cs
using PAR.Domain.Common;

namespace PAR.Domain.Entities;

public class SchoolYear : BaseEntity
{
     public string Name { get; set; } = null!;
     public DateOnly StartDate { get; set; }
     public DateOnly EndDate { get; set; }
     public bool IsCurrent { get; set; }
     public virtual ICollection<Group> Groups { get; set; } = new HashSet<Group>();
}

File: src/par-app/src/app/models/schoolYear.ts
import {BaseDto} from "./common/baseDto";
import {GroupDto} from "./group";

export interface SchoolYearDto extends BaseDto {
    name: string;
    startDate: Date;
    endDate: Date;
    isCurrent: boolean;
    groups: GroupDto[];
}

export class SchoolYearFormValues {
    id: number = 0;
    name: string = '';
    startDate: Date = new Date();
    endDate: Date = new Date();
    isCurrent: boolean = false;

    constructor(schoolYear?: SchoolYearDto) {
        if (schoolYear) {
            this.id = schoolYear.id;
            this.name = schoolYear.name;
            this.startDate = schoolYear.startDate;
            this.endDate = schoolYear.endDate;
            this.isCurrent = schoolYear.isCurrent;
        }
    }
}